### PR TITLE
Add a molecule configuration file for SystemD-enabled Docker images

### PR DESCRIPTION
**Pull request by jsf9k**
## 🗣 Description ##

This pull request:
* Renames the existing `molecule.yml` file to `molecule-no-systemd.yml`
* Adds a molecule configuration file called `molecule-with-systemd.yml` for SystemD-enabled Docker images
* Adds a symlink called `molecule.yml` that points to `molecule-no-systemd.yml`

## 💭 Motivation and Context ##

We have several Ansible role repositories that require a SystemD-enabled molecule configuration.  So far we have been modifying the `molecule.yml` file to use the SystemD-enabled Docker images separately in each of those repositories.  This is obviously prone to error and difficult to maintain; instead, we should be somehow managing the SystemD-enabled molecule configuration centrally.  This PR achieves that.

I did consider creating a separate skeleton repository for this use case - something like cisagov/skeleton-ansible-role-with-systemd - but there are several cases where we require both a SystemD-enabled molecule _and_  an AWS test user.  In those cases it would be ambiguous whether one should use the cisagov/skeleton-ansible-role-with-systemd or [cisagov/skeleton-ansible-role-with-test-user](https://github.com/cisagov/skeleton-ansible-role-with-test-user) skeleton; furthermore, a separate skeleton seems like a lot of overhead to change a single file.  I think the solution in this PR is superior.

## 🧪 Testing ##

All pre-commit hooks and molecule tests pass.

## ✅ Checklist ##

* [x] This PR has an informative and human-readable title.
* [x] Changes are limited to a single goal - _eschew scope creep!_
* [x] All relevant type-of-change labels have been added.
* [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
* [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
* [x] All new and existing tests pass.

### CODE REVIEW

File: molecule/default/molecule-with-systemd.yml [hunk 1]
```diff
@@ -0,0 +1,123 @@
...
+  config_options:
+    defaults:
+      # The OS family and distribution for Kali is "Kali GNU/Linux",
+      # which is not an allowable group name in Ansible due to the
+      # space and the slash.  This option converts those characters to
+      # underscores when creating the group name.
+      force_valid_group_names: always
+  inventory:
+    host_vars:
+      debian9:
```
Comment (mcdonnnj) [inline molecule/default/molecule-with-systemd.yml]:
```suggestion
      debian9_systemd:
```

**Review by mcdonnnj — APPROVED** [review]
I like this solution and it makes it a lot easier to start a new role that needs `systemd`-enabled images. I had one fix as a suggestion.

File: molecule/default/molecule-with-systemd.yml (hunk 1 above)
Comment (jsf9k) [reply molecule/default/molecule-with-systemd.yml]: Good catch!

**Review by hillaryj — APPROVED** [review]
Great solution!

**Review by dav3r — APPROVED** [review]
This is excellent! 🥇 

My only feedback is that I think we should put a brief comment at the top of `molecule-no-systemd.yml` and `molecule-with-systemd.yml` to explain how/when each file should be used.

**Comment by jsf9k** [comment]
When the Kraken containing this PR is released, we should also remove any `regex_replace()` foo that we can from the Ansible role repositories.  That jazz should no longer be necessary since Ansible now correctly identifies Kali Linux as belonging to the Debian Linux family.

**Comment by jsf9k** [comment]
> My only feedback is that I think we should put a brief comment at the top of `molecule-no-systemd.yml` and `molecule-with-systemd.yml` to explain how/when each file should be used.

@dav3r - see commit f692024.

**Comment by dav3r** [comment]
> > My only feedback is that I think we should put a brief comment at the top of `molecule-no-systemd.yml` and `molecule-with-systemd.yml` to explain how/when each file should be used.
> 
> @dav3r - see commit [f692024](https://github.com/cisagov/skeleton-ansible-role/commit/f692024d75a34623c38491cab3ccaa9473c3d53e).

That looks great to me- thanks for adding that!

**Pull request merged**